### PR TITLE
Document DKIM support and fix permissions bug

### LINKED ISSUE #26: Document DKIM_KEY support, fix DKIM permission bug
See https://github.com/namshi/docker-smtp/issues/22#issuecomment-762485772.

If we see the DKIM key, the owner needs to be changed to `Debian-exim`.

### PR DESCRIPTION
**Pull request by yanokwa**
Closes #26.

I've confirmed that the instructions and bash if block works on my container, but I have not tested a fresh end-to-end configuration.

### CODE REVIEW

**Comment by tlex** [comment]
Thank you for the pull request.

I particularly appreciate the enrichment of the documentation.

I would prefer however, instead of changing permissions on the fly, to have the file directly with the right permissions.

For this to work, the file should have group id `101` and permissions `640` (assuming `userns` is disabled):
```
root@4230594df42e:/# id Debian-exim
uid=101(Debian-exim) gid=101(Debian-exim) groups=101(Debian-exim)
```

My suggestion, therefore, is to add two extra steps to the documentation (after generating the keys):

```sh
chown :101 rsa.private
chmod 640 rsa.private
```

**Comment by yanokwa** [comment]
Thanks for the fast review and for maintaining this useful container! 

I had considered the approach you suggested and decided against it because openssl writes rsa.private with 600 (as recommended at http://linuxcommand.org/lc3_man_pages/ssh1.html). Given that, weakening the permissions outside the container seems unnecessarily less secure, no?

**Comment by tlex** [comment]
Hi again,

Thinking about this:

> Given that, weakening the permissions outside the container seems unnecessarily less secure, no?

The approach with the bind mount will actually change the permissions of the file on disk, if implemented like this.

I'll make a suggestion to the code, to handle this differently. Basically, if a certain file exists at start, to copy it with the right permissions inside the container.

File: README.md [hunk 1]
```diff
@@ -85,6 +85,39 @@ services:
...
+```
+
+Finally, customize your `docker-compose.yml` to mount the necessary files. In this example, we've put the files on the host at `./config/ixdotai-smtp`.
+
+```yml
+mail:
+  image: "ixdotai/smtp"
+  volumes:
+    - ./config/ixdotai-smtp/config:/etc/exim4/_docker_additional_macros:ro
+    - ./config/ixdotai-smtp/rsa.private:/etc/exim4/domain.key
```
Comment (tlex) [inline README.md:116]:
```suggestion
    - ./config/ixdotai-smtp/rsa.private:/etc/exim4/private.key
```

File: entrypoint.sh [hunk 2]
```diff
@@ -111,6 +111,12 @@ else
 	)
 fi
 
+# allow Debian-exim user to read domain key for DKIM support
+if [ -f /etc/exim4/domain.key ]; then
+	chgrp Debian-exim /etc/exim4/domain.key
+	chmod 640 /etc/exim4/domain.key
+fi
```
Comment (tlex) [inline entrypoint.sh:118]:
```suggestion
if [ -f /etc/exim4/private.key ]; then
	cp /etc/exim4/private.key /etc/exim4/domain.key
	chown :101 /etc/exim4/domain.key
	chmod 640 /etc/exim4/domain.key
fi
```

File: entrypoint.sh [hunk 3]
```diff
@@ -60,6 +60,19 @@ else
 	  echo "MAIN_TLS_ADVERTISE_HOSTS = !*" >> /etc/exim4/exim4.conf.localmacros
 fi
 
+if [ "$DKIM_KEY_PATH" ]; then
+	cp "$DKIM_KEY_PATH" /etc/exim4/dkim.key
+	chown :101 /etc/exim4/dkim.key
+	chmod 640 /etc/exim4/dkim.key
+	{
```
Comment (yanokwa) [inline entrypoint.sh:67]: Recommended by https://www.shellcheck.net/wiki/SC2129

File: entrypoint.sh [hunk 4]
```diff
@@ -60,6 +60,19 @@ else
 	  echo "MAIN_TLS_ADVERTISE_HOSTS = !*" >> /etc/exim4/exim4.conf.localmacros
 fi
 
+if [ "$DKIM_KEY_PATH" ]; then
```
Comment (yanokwa) [inline entrypoint.sh:63]: Trying to match existing pattern of KEY_PATH

**Comment by yanokwa** [comment]
I've made some further changes to make it easier for users to implement. Just one mount and one ENV variable. 

I don't love appending `.temp` to the name, but it at least gives users a sense that the file isn't used as is.

**Comment by tlex** [comment]
Thank you again, I'm quite happy with the solution.

**Comment by tlex** [comment]
Released in v0.5.0

**Pull request closed without merging**